lib/base58encode.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Base58Encode do
+  @moduledoc """
+  Provides the encode function which takes a binary and return it's
+  reprentation in base58
+  """
+
+  @doc """
+  ## Examples
+
+    iex> Base58Encode.encode("hello")
+    "Cn8eVZg"
+
+    iex> Base58Encode.encode(42)
```

Comment:
README.md says that `encode(42)` is `"bQbp"` https://github.com/dwyl/base58encode/pull/3/files#r251960830

Reply:
The Readme was wrong! I've update the Readme to try to explain the difference between Elixir binary and binary numbers.